automation/src/main/java/org/wso2/testgrid/automation/executor/JMeterExecutor.java

```diff
@@ -92,7 +94,7 @@ public void execute(String script, Deployment deployment) throws TestAutomationE
             throw new TestAutomationException(StringUtil.concatStrings("Script file ", script, " cannot be located."));
         }
 
-        ResultCollector resultCollector = new JMeterResultCollector(summariser, testScenario);
+        JMeterResultCollector resultCollector = new JMeterResultCollector(summariser, testScenario);
```

Comment:
We can set the test cases to the testScenario and persist testScenario directly. This will persist all the test cases. However please do check if we are getting a null pointer exception stating that the entity ID is null.

Reply:
+1. Not required for this PR. Please raise a ticket.

Reply:
Raised #474